include/envoy/server/overload_manager.h

```diff
@@ -29,6 +30,23 @@ class OverloadActionState {
 
   float value() const { return action_value_; }
   bool isSaturated() const { return action_value_ == 1; }
+  bool isInactive() const { return action_value_ == 0; }
+
+  /**
+   * Converts the overload action state to a boolean. The return value will be true with probability
+   * of the level of saturation.
+   */
+  bool isRandomizedActive(Random::RandomGenerator& random_generator) const {
+    if (isInactive()) {
```

Comment:
Is this just for performance, or do the edges not work well with the casts?

Reply:
This was stolen from #13120, but I assume it's for performance - don't grab a random number if it's not necessary. That's why this PR is still a draft.